### PR TITLE
Add premerge integration tests for getAddresses

### PR DESCRIPTION
**Pull request by MattBidewell**
### What changed

Add premerge acceptance tests to the repo for getAddresses lambda function. 

### Why did it change

This will give us great confidence changes to the lambda will continue to work.

### CODE REVIEW

File: integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java [hunk 1]
```diff
@@ -322,6 +324,29 @@ public void aValidJWTIsReturnedInTheResponse() throws ParseException, IOExceptio
...
+        this.response = sendHttpRequest(request);
+    }
+
+    @Then("user should receive a valid response")
+    public void userShouldReceiveAValidResponse() throws ParseException, IOException {
+        assertEquals(200, response.statusCode());
+        var payload = objectMapper.readTree(response.body().toString());
+
+        assertEquals(
+                "MK42 7AA",
```
Comment (MattBidewell) [inline integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java:346]: Not sure where this is being set from. I logged out the response and it had a postcode set, but a global search of where this postcode came from resulted in nothing...

Comment (techaman-m) [reply integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java:345]: The postcode is actually fetched from the user identity which we get in the initial steps of the test [here](https://github.com/alphagov/di-ipv-cri-address-api/blob/19489cb916410b2822e25c90a4181871ed86e5ab/integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java#L71)

You can pick the json object from the above step to find the postcode and assert that here.
E.g.
`
    Map<String, String> deserializedResponse =
                objectMapper.readValue(userIdentityJson, new TypeReference<>() {});
        String inputPostCode = deserializedResponse.get("addresses").get("postcode");
`

and in test:
 assertEquals(
                inputPostCode, payload.get("result").get("addresses").get(0).get("postalCode").asText());

File: integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java [hunk 2]
```diff
@@ -322,6 +323,29 @@ public void aValidJWTIsReturnedInTheResponse() throws ParseException, IOExceptio
         makeAssertions(SignedJWT.parse(response.body()));
     }
 
+    @When("use sends a GET request to Addresses end point")
```
Comment (techaman-m) [inline integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java:326]: typo for word `user`

File: integration-tests/src/test/resources/features/AddressAPIHappyPath.feature [hunk 3]
```diff
@@ -27,3 +27,7 @@ Feature: Address API happy path test
     When user sends a POST request to Credential Issue end point with a valid access token
     And a valid JWT is returned in the response
 
+    #Get_Addresses
+    When use sends a GET request to Addresses end point
```
Comment (techaman-m) [inline integration-tests/src/test/resources/features/AddressAPIHappyPath.feature:31]: typo in word `user`

**Review by techaman-m — APPROVED** [review]

**Review by mrdoodles — APPROVED** [review]

**Pull request merged**